specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2020-03-01-preview/componentLinkedStorageAccounts_API.json

```diff
@@ -0,0 +1,299 @@
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StorageTypeParameter"
+          },
+          {
+            "name": "LinkedStorageAccountsProperties",
+            "description": "Properties that need to be specified to update linked storage accounts for an Application Insights component.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ComponentLinkedStorageAccounts"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request when updating linked storage accounts for an Application Insights component. The updated linked storage account is returned.",
+            "schema": {
+              "$ref": "#/definitions/ComponentLinkedStorageAccounts"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentLinkedStorageAccountsCreateAndUpdate": {
+            "$ref": "./examples/LinkedStorageAccountsCreateAndUpdate.json"
+          }
+        }
+      },
+      "patch": {
+        "description": "Update linked storage accounts for an Application Insights component.",
+        "operationId": "ComponentLinkedStorageAccounts_Update",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/StorageTypeParameter"
+          },
+          {
+            "name": "LinkedStorageAccountsProperties",
+            "description": "Properties that need to be specified to update linked storage accounts for an Application Insights component.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ComponentLinkedStorageAccounts"
```

Comment:
Patch should only allow individually writable properties. The entirety of the Put payload shouldn't be used.

Please create a new definition called ComponentLinkedStorageAccountsUpdate which contains the patchable fields.

Reply:
Fixed